Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -891,6 +894,7 @@ public bool UnregisterPointer(IMixedRealityPointer pointer)
                 }
 
                 pointers.Remove(pointerData.Pointer.PointerId);
+                pointersList.Remove(pointerData);
```

Comment:
I'm almost never a fan of tracking the same thing in multiple places...what's the optimization here?

Reply:
The issue is, without the list you have to foreach on a Dictionary's keys, which allocates. This would be OK if the allocation didn't happen on every frame multiple times.